src/common/configuration.h

```diff
@@ -0,0 +1,184 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the Apache 2.0 License.
+
+#pragma once
+
+#include "common/enclave_interface_types.h"
+#include "consensus/consensus_types.h"
+#include "crypto/curve.h"
+#include "ds/logger.h"
+#include "ds/oversized.h"
+#include "enclave/consensus_type.h"
+#include "enclave/reconfiguration_type.h"
+#include "node/config.h"
+#include "node/members.h"
+#include "node/node_info_network.h"
+
+#include <optional>
+#include <string>
+#include <vector>
+
+namespace logger
+{
+#ifdef VERBOSE_LOGGING
+  DECLARE_JSON_ENUM(
+    Level,
+    {{Level::TRACE, "trace"},
+     {Level::DEBUG, "debug"},
+     {Level::INFO, "info"},
+     {Level::FAIL, "fail"},
+     {Level::FATAL, "fatal"}});
+#else
+  DECLARE_JSON_ENUM(
+    Level,
+    {{Level::INFO, "info"}, {Level::FAIL, "fail"}, {Level::FATAL, "fatal"}});
+#endif
+}
+
+struct SizeString
+{
+  size_t value;
+
+  SizeString() = default;
+  SizeString(size_t val) : value(val) {}
+
+  bool operator==(const SizeString&) const = default;
+
+  inline operator size_t() const
+  {
+    return value;
+  }
+};
+
+// Note: from_json() is defined differently whether for host or in enclave
```

Comment:
It's a little confusing that this isn't next to from_json or that it doesn't explain why we do this (to avoid cli11 in enclave I'm guessing).